### PR TITLE
A first try of openrefine GIE -> interactivetool

### PR DESCRIPTION
**Pull request by yvanlebras**
Here is a first try to take an ancient Openrefine dockerized GIE and convert it to an interactivetool. I don't have possibility to test it on a test Galaxy instance for now, if someone from eu can try and validate or not....

### CODE REVIEW

File: tools/interactive/interactivetool_openrefine.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                #for $api_key in $__user__.api_keys:
+                    ${api_key.key}
+                    #break
+                #end for
+            #end if
+        </environment_variable> <!-- FIXME: We should have a better way to get user's API key -->
+    </environment_variables>
+    <command><![CDATA[
+
+        /startup.sh
```
Comment (bgruening) [inline tools/interactive/interactivetool_openrefine.xml]: Thats all? Cool :)

File: tools/interactive/interactivetool_openrefine.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            <url></url>
+        </entry_point>
+    </entry_points>
+    <environment_variables>
+
+        <environment_variable name="HISTORY_ID" strip="True">${__app__.security.encode_id($outfile.history_id)}</environment_variable> <!-- FIXME: Warning: The use of __app__ is deprecated and will break backward compatibility in the near future -->
+        <environment_variable name="REMOTE_HOST">${__app__.config.galaxy_infrastructure_url}</environment_variable> <!-- FIXME: Warning: The use of __app__ is deprecated and will break backward compatibility in the near future -->
+        <environment_variable name="GALAXY_WEB_PORT">8080</environment_variable>
+        <environment_variable name="GALAXY_URL">${__app__.config.galaxy_infrastructure_url}</environment_variable> <!-- FIXME: Warning: The use of __app__ is deprecated and will break backward compatibility in the near future -->
+        <environment_variable name="API_KEY" strip="True">
```
Comment (bgruening) [inline tools/interactive/interactivetool_openrefine.xml]: @yvanlebras there is a new method to do this, please check out the new juypter one.

Comment (bgruening) [reply tools/interactive/interactivetool_openrefine.xml]: https://github.com/usegalaxy-eu/galaxy/blob/release_19.09_europe/tools/interactive/interactivetool_jupyter_notebook.xml#L15

**Comment by bgruening** [comment]
Anything special in your container? Or can we use https://hub.docker.com/r/felixlohmeier/openrefine/ ???

**Comment by yvanlebras** [comment]
In fact, in our container, there is use of Galaxy ie helper so we can import and export dataset from/to Galaxy history... Unfortunately, I am trying things since few hours (commenting lines related to Galaxy connection so I can deploy a docker image) and encounter a lot of issues so not sure that this interactivetool can work like this... notably due to a internal container nginx who was used on GIE to look at `PROXY_PREFIX` and redirecting from http://127.0.0.1:3333/ to PROXY_PREFIX/openrefine/ ... But it seems to me this was specific to GIE (If I well remember, for GIE, PROXY_PREFIX was not known before ontainer execution isn't it ?) and not to consider for interactivetool.... So I think I maybe have to change the actual proxy.conf but don't succeed to make something working....
```
server {
    listen 80;
    server_name localhost;
    access_log /var/log/nginx/localhost.access.log;

    location PROXY_PREFIX/openrefine/ {
        proxy_buffering off;
        proxy_pass         http://127.0.0.1:3333/;
        proxy_redirect     http://127.0.0.1:3333/ PROXY_PREFIX/openrefine/;
    }
}

```

**Comment by bgruening** [comment]
Oh interesting ... what is the use-case for get/put?

@yvanlebras I think you can remove the entire nginx.

**Comment by yvanlebras** [comment]
exactly what I was thinking about.... but manipulating proxy is really really like crossing highroad for me ;) But I continue to investigate the code https://github.com/yvanlebras/openrefine-docker

**Comment by bgruening** [comment]
What I mean is, remove nginx completely and then use the port of openrefine (http://127.0.0.1:3333) ... that should maybe also work.

**Comment by yvanlebras** [comment]
yes, agree. I am trying ;)

**Comment by bgruening** [comment]
please merge https://github.com/yvanlebras/openrefine-docker/pull/1 and then I will proceed here ..

**Comment by bgruening** [comment]
Ok, export also works. At least it is transferring files to Galaxy. However, no matter what I choose, export Excel, XML ... its always tsv ... this is confusing, but in theory this works.

**Comment by yvanlebras** [comment]
merge done!

**Pull request merged**